provider/ai_task.go

```diff
@@ -21,13 +22,43 @@ type AITaskSidebarApp struct {
 // TaskPromptParameterName is the name of the parameter which is *required* to be defined when a coder_ai_task is used.
 const TaskPromptParameterName = "AI Prompt"
 
-func aiTask() *schema.Resource {
+func aiTaskResource() *schema.Resource {
 	return &schema.Resource{
 		SchemaVersion: 1,
 
 		Description: "Use this resource to define Coder tasks.",
 		CreateContext: func(c context.Context, resourceData *schema.ResourceData, i any) diag.Diagnostics {
-			resourceData.SetId(uuid.NewString())
+			if idStr := os.Getenv("CODER_TASK_ID"); idStr != "" {
+				resourceData.SetId(idStr)
+			} else {
+				resourceData.SetId(uuid.NewString())
+			}
```

Comment:
Would it make more sense to exit with an error here?

Reply:
The integration tests will tell us :)
I'd expect that it won't have any effect unless the template defines a `coder_ai_task`. In that case, folks can lock their provider version.